check.ts

```diff
@@ -0,0 +1,140 @@
+      //   });
+      //   return res.status(200).json({ message: 'Repository not configured for review' });
+      // }
+
+      this.processReviewAsync(repository.owner.login, repository.name, number);
+    } catch (error: any) {
+      logger.error('Error handling PR webhook:', { error: error.message });
+      res.status(500).json({ error: 'Internal server error' });
+    }
+  }
+
+  private async processReviewAsync(owner: string, repo: string, pullNumber: number): Promise<void> {
+    try {
+      logger.info('Starting async PR review', { owner, repo, pullNumber });
+
+      const result = await this.reviewService.reviewPullRequest(owner, repo, pullNumber);
+
+      logger.info('Async PR review completed', {
+        owner,
+        repo,
+        pullNumber,
+        status: result.status,
+      });
+    } catch (error: any) {
+      logger.error('Error in async PR review:', {
+        error: error.message,
+        stack: error.stack,
+        owner,
+        repo,
+        pullNumber,
+      });
+    }
+  }
+
+  async handlePullRequestReview(req: Request<{}, {}, PullRequestReviewWebhookEvent>, res: Response): Promise<void> {
+    try {
+      const { action, review, pull_request, repository } = req.body;
+
+      logger.info('Received PR review webhook', {
+        action,
+        reviewState: review?.state,
+        prNumber: pull_request?.number,
+        repo: repository?.full_name,
+      });
+
+      // For now, we'll just log review events
+      // Could be extended to update Linear issues based on review outcomes
+
+      res.status(200).json({ message: 'Review webhook received' });
+    } catch (error: any) {
+      logger.error('Error handling PR review webhook:', { error: error.message });
+      res.status(500).json({ error: 'Internal server error' });
+    }
+  }
+
+  async handleIssueComment(req: Request<{}, {}, IssueCommentWebhookEvent>, res: Response): Promise<void> {
+    try {
+      const { action, comment, issue, repository } = req.body;
+
+      // Only process if it's a PR comment (issues and PRs share the same endpoint)
+      if (!issue.pull_request) {
+        res.status(200).json({ message: 'Not a PR comment' });
+        return;
+      }
+
+      logger.info('Received issue comment webhook', {
+        action,
+        commentId: comment?.id,
+        issueNumber: issue?.number,
+        repo: repository?.full_name,
+      });
+
+      // Could be extended to handle commands in comments (e.g., "/review", "/security-check")
+
+      res.status(200).json({ message: 'Comment webhook received' });
+    } catch (error: any) {
+      logger.error('Error handling issue comment webhook:', { error: error.message });
+      res.status(500).json({ error: 'Internal server error' });
+    }
+  }
+
+  async healthCheck(_req: Request, res: Response): Promise<void> {
+    try {
+      const health = {
+        status: 'healthy',
+        timestamp: new Date().toISOString(),
```

Comment:
Direct environment variable access violates the config pattern established in this file. Use the config module for consistency. ```suggestion
        version: config.APP_VERSION || '1.0.0',
```